list-agreements.lua

```diff
@@ -1,120 +1,218 @@
 end
 
 function get_location_type(agr)
     loctype = agr.details[0].data.Location.type
     return loctype
 end
 
-function get_location_name(agr)
-    if get_location_type(agr) == df.abstract_building_type.TEMPLE and get_location_tier(agr) == 1 then
-        return "temple"
-    elseif get_location_type(agr) == df.abstract_building_type.TEMPLE and get_location_tier(agr) == 2 then
-        return "temple complex"
-    elseif get_location_type(agr) == df.abstract_building_type.GUILDHALL and get_location_tier(agr) == 1 then
-        return "guildhall"
-    elseif get_location_type(agr) == df.abstract_building_type.GUILDHALL and get_location_tier(agr) == 2 then
-        return "grand guildhall"
-    end
+function get_petition_date(agr)
+    local agr_year = agr.details[0].year
+    local agr_year_tick = agr.details[0].year_tick
+    local julian_day = math.floor(agr_year_tick / 1200) + 1
+    local agr_month = math.floor(julian_day / 28) + 1
+    local agr_day = julian_day % 28
+    return string.format("%03d-%02d-%02d",agr_year, agr_month, agr_day)
 end
 
-function get_petition_date(agr)
-    agr_year = agr.details[0].year
-    agr_year_tick = agr.details[0].year_tick
-    julian_day = math.floor(agr_year_tick / 1200) + 1
-    agr_month = math.floor(julian_day / 28) + 1
-    agr_day = julian_day % 28
-    return agr_year.."-"..agr_month.."-"..agr_day
+function get_petition_age(agr)
+    local agr_year_tick = agr.details[0].year_tick
+    local agr_year = agr.details[0].year
+    local cur_year_tick = df.global.cur_year_tick
+    local cur_year = df.global.cur_year
+    local del_year, del_year_tick, del_month, del_day
+    --delta, check to prevent off by 1 error, not validated
+    if cur_year_tick > agr_year_tick then
+        del_year = cur_year - agr_year
+        del_year_tick = cur_year_tick - agr_year_tick
+    else
+        del_year = cur_year - agr_year - 1
+        del_year_tick = agr_year_tick - cur_year_tick
+    end
+    local julian_day = math.floor(del_year_tick / 1200) + 1
+    del_month = math.floor(julian_day / 28)
+    del_day = julian_day % 28
+    return {del_year,del_month,del_day}
 end
 
 function get_guildhall_profession(agr)
-    prof = agr.details[0].data.Location.profession
-    profname = string.lower(df.profession[prof])
+    local prof = agr.details[0].data.Location.profession
+    local profname = string.lower(df.profession[prof])
+    -- *VERY* important code follows
+    if string.find(profname, "man") then
+        profname = string.gsub(profname,"man",string.lower(dfhack.units.getRaceNameById(df.global.ui.race_id)))
+    end
     return profname:gsub("_", " ")
 end
 
-function get_religion_name(agr)
-    religion_id = agr.details[0].data.Location.deity_data.Religion
-    religion_name = dfhack.TranslateName(df.global.world.entities.all[religion_id].name, true)
-    return religion_name
+function get_agr_party_name(agr)
+    --assume party 0 is guild/order, 1 is local government as siteid = playerfortid
```

Comment:
is this an assumption or is it documented somewhere?

Reply:
inference based on documentation and preceding code

relevant structure definition: [df-structures/df.entities L1471](https://github.com/DFHack/df-structures/blob/0fee473dbca37e6d3fd99d2e7447d68418d1158f/df.entities.xml#L1471) 

we've already filtered down to only this sub-type of agreement (location), and filtered out agreements made at other world sites (line 177: `if agr.details[0].data.Location.site == playerfortid then`); so therefore, the 'government' `party[1]` as documented, MUST be the player, and applicant `party[0]`, it being a location agreement will be either a guild or religion (not withstanding toady introducing theatres for performance troupes or other features)

I can see a scenario in which these could appear in arbitrary order, which would be easy enough to handle by testing `for _,party_id in pairs...` `is party_id = playergovernment_id` but i think my logic above is satisfactory as the arrays are strictly typed (unlike, for example JSON in which headers can appear in any order)